DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/BulkUploadController.cs

```diff
@@ -95,16 +95,24 @@ public IActionResult StartUpload(UploadDelegatesViewModel model)
             {
                 return View("Index", model);
             }
-            var workbook = new XLWorkbook(model.DelegatesFile.OpenReadStream());
-            if (!workbook.Worksheets.Contains(DelegateDownloadFileService.DelegatesSheetName))
+            try
             {
-                ModelState.AddModelError("MaxBulkUploadRows", CommonValidationErrorMessages.InvalidBulkUploadExcelFile);
-                return View("StartUpload", model);
-            }
-            var delegateFileName = FileHelper.UploadFile(webHostEnvironment, model.DelegatesFile);
-            setupBulkUploadData(centreId, adminUserID, delegateFileName);
+                var workbook = new XLWorkbook(model.DelegatesFile.OpenReadStream());
+                if (!workbook.Worksheets.Contains(DelegateDownloadFileService.DelegatesSheetName))
+                {
+                    ModelState.AddModelError("MaxBulkUploadRows", CommonValidationErrorMessages.InvalidBulkUploadExcelFile);
+                    return View("StartUpload", model);
+                }
+                var delegateFileName = FileHelper.UploadFile(webHostEnvironment, model.DelegatesFile);
+                setupBulkUploadData(centreId, adminUserID, delegateFileName);
 
-            return RedirectToAction("UploadComplete");
+                return RedirectToAction("UploadComplete");
+            }
+            catch (Exception)
+            {
+                ModelState.AddModelError("DelegatesFile", "The Excel file has at least one cell containing an invalid hyperlink or email address.");

```

Comment:
That's an excellent catch! I've changed the code to catch the exact error that we are expecting because of the malformed URI and display the error message accordingly.